bcipy/task/base_calibration.py

```diff
@@ -0,0 +1,324 @@
+        if target == symbol:
+            return TriggerType.TARGET
+        return TriggerType.NONTARGET
+
+    def present_inquiry(self, index: int,
+                        inquiry: Inquiry) -> List[Tuple[str, float]]:
+        """Present the given inquiry and return the trigger timing info.
+
+        Parameters
+        ----------
+        - index : current index
+        - inquiry : next sequence of stimuli to present.
+
+        Returns
+        -------
+        List of stim_times (tuples representing the stimulus and time that
+            it was presented relative to the experiment clock)
+        """
+        self.display.update_task_bar(str(index + 1))
+        self.display.draw_static()
+        self.window.flip()
+
+        self.display.schedule_to(inquiry.stimuli, inquiry.durations,
+                                 inquiry.colors)
+        self.wait()
+        timing = self.display.do_inquiry()
+        return timing
+
+    def user_wants_to_continue(self, first_inquiry: bool = False) -> bool:
+        """Check if user wants to continue or terminate.
+
+        Returns
+        -------
+        - `True` to continue
+        - `False` to finish the task.
+        """
+        should_continue = get_user_input(self.display,
+                                         self.wait_screen_message,
+                                         self.wait_screen_message_color,
+                                         first_run=first_inquiry)
+        if not should_continue:
+            self.logger.info('User wants to exit.')
+        return should_continue
+
+    def execute(self) -> str:
+        """Task run loop."""
+        self.logger.info(f'Starting {self.name()}!')
+        self.wait()
+
+        inq_index = 0
+        while self.user_wants_to_continue(inq_index == 0):
+            try:
+                inquiry = next(self.inquiry_generator)
+            except StopIteration:
+                break
+            if self.enable_breaks:
+                # Take a break at the defined interval
+                pause_calibration(self.window, self.display, inq_index,
+                                  self.parameters)
+
+            timing = self.present_inquiry(inq_index, inquiry)
+
+            # Write triggers for the inquiry
+            self.write_trigger_data(timing, inq_index == 0)
+            self.add_session_data(inquiry)
+
+            # Wait for a time
+            self.wait()
+            inq_index += 1
+
+        self.exit_display()
+        self.write_offset_trigger()
+        self.cleanup()
+
+        return self.file_save
+
+    def exit_display(self) -> None:
+        """Close the UI and cleanup."""
+        # Say Goodbye!
+        self.display.info_text = trial_complete_message(
+            self.window, self.parameters)
+        self.display.draw_static()
+        self.window.flip()
+
+        # Allow for some additional data to be collected for later processing
+        self.wait()
+
+    def cleanup(self) -> None:
```

Comment:
This could throw a not implemented error. We can add that in the future once we add cleanup to these tasks more explicitly though.

Reply:
Yeah, I think cleanup may be optional for many tasks so I prefer to leave this empty by default.